src/core/weights.js

```diff
@@ -48,19 +57,72 @@ export function copy(w: Weights): Weights {
   };
 }
 
-/**
- * Merge multiple Weights together.
+/** Merge multiple Weights together.
+ *
+ * The resultant Weights will have every weight specified by each of the input
+ * weights.
+ *
+ * When there are overlaps (i.e. the same address is present in two or more of
+ * the Weights), then the appropriate resolver will be invoked to resolve the
+ * conflict. The semantics are similar to Array.reduce, where the accumulated
+ * weight and the current weight are both passed as arguments to the resolver.
+ * For the first conflict, the "accumulated weight" is equal to the first
+ * weight matching that address. The final accumulated weight will be included
+ * in the resultant Weights.
  *
- * The resultant Weights will have every weight specified by each of the
- * input weights. If there are any overlaps (i.e. the same address is present
- * in two or more of the input weights), an error will be thrown. In the future,
- * we will likely modify this function to add a resolver that determines how to
- * combine multiple overlapping weights.
+ * When no resolvers are explicitly provided, the merge defaults to
+ * conservative "error on conflict" resolvers.
  */
-export function merge(ws: $ReadOnlyArray<Weights>): Weights {
-  const nodeWeights = MapUtil.merge(ws.map((x) => x.nodeWeights));
-  const edgeWeights = MapUtil.merge(ws.map((x) => x.edgeWeights));
-  return {nodeWeights, edgeWeights};
+export function merge(
+  ws: $ReadOnlyArray<Weights>,
+  nodeResolver: ?NodeOperator,
+  edgeResolver: ?EdgeOperator
```

Comment:
Optional: Consider passing these as an optional object with optional
keys, so that the parameters can be named? `merge(ws, null, null)` is a
tad awkward.

Reply:
Agree :] and I would ask this a bit more strongly than optional.
Would argue not doing so adds a bit of API technical debt.

Over time a practice of multiple optional positional arguments makes the invocations brittle and really hard to glean the function signature. Named parameters help out a lot there.

`createReverseIndex(articles, null, 3);`
:thinking: what do you mean 3? How do I know this 3 shouldn't be in the 2nd position? _(Answer: need to look up the definition)_.

`createReverseIndex(articles, {minTokenLength: 3});`
Ah!

Reply:
Done.

Reply:
@Beanow: Yes; `JSON.stringify(obj, null, 4)` is a commonplace example.